prime-router/src/main/kotlin/SubmissionReceiver.kt

```diff
@@ -262,6 +263,21 @@ class ELRReceiver : SubmissionReceiver {
             report, rawBody, sender, actionHistory, payloadName
         )
 
+        // check for valid message type
+        for (message in messages) {
+            val header = message.get("MSH")
+            check(header is MSH)
+            val messageType = header.messageType.msg1_MessageCode.value +
+                "_" +
+                header.messageType.msg2_TriggerEvent.value
+
+            // TODO: This may need to be a configurable value in the future, if we ever support message types other
+            //  than ORU_RO1. As of 6/15/2022 multiple message type support is out of scope
+            if (messageType != "ORU_R01") {
+                throw IllegalArgumentException("Ignoring unsupported HL7 message type $messageType")
```

Comment:
Change this to an ActionLog message to be the same as the one used in HL7 serializer.  This will be caught at line 282 with any other errors that may be added in the future.  Note this message is also a warning in the COVID pipeline, so if we want to do the same then you need to filter messages out.
```
warnings.add(
                        InvalidHL7Message
                        ("Ignoring unsupported HL7 message type ${msgType.joinToString(",")}")
                    )
```